src/units/abilities.cpp

```diff
@@ -1120,6 +1120,42 @@ void attack_type::modified_attacks(bool is_backstab, unsigned & min_attacks,
 	}
 }
 
+unit_ability_list attack_type::get_specials_and_abilities_with_string(const std::string& special) const
+{
+	//the custom_priority attribute is used for define rules of priority of damage type
+	unit_ability_list abil_list = get_specials_and_abilities(special);
+	unit_ability_list temp_list;
+	int priority = 1;
+	if(abil_list.empty()){
+		return abil_list;
+	}
+	while(priority>=1 && priority<=100){
+		for(const auto& i : abil_list) {
+			int temp_priority = std::clamp((*i.ability_cfg)["type_priority"].to_int(1), 1, 100);
```

Comment:
And this will need to be changed to… I think it's called `as_double`?

Reply:
double, it's done